domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs

```diff
@@ -0,0 +1,138 @@
+    /// <response code="201">Created</response>
+    /// <response code="409">There is already a Transfer Agreement with this company tin within the selected date range</response>
+    [ProducesResponseType(typeof(Guid), 201)]
+    [ProducesResponseType(typeof(string), 409)]
+    [HttpPost]
+    public async Task<ActionResult> CreateTransferAgreementProposal(CreateTransferAgreementProposal request)
+    {
+        var companySenderId = Guid.Parse(User.FindSubjectGuidClaim());
+        var companySenderTin = User.FindSubjectTinClaim();
+        var companySenderName = User.FindSubjectNameClaim();
+
+        var validateResult = await createTransferAgreementProposalValidator.ValidateAsync(request);
+        if (!validateResult.IsValid)
+        {
+            validateResult.AddToModelState(ModelState);
+            return ValidationProblem(ModelState);
+        }
+
+        var newProposal = new TransferAgreementProposal
+        {
+            SenderCompanyId = companySenderId,
+            SenderCompanyTin = companySenderTin,
+            SenderCompanyName = companySenderName,
+            Id = Guid.NewGuid(),
+            ReceiverCompanyTin = request.ReceiverTin,
+            StartDate = DateTimeOffset.FromUnixTimeSeconds(request.StartDate),
+            EndDate = request.EndDate == null ? null : DateTimeOffset.FromUnixTimeSeconds(request.EndDate.Value)
+        };
+
+        var hasConflict = await transferAgreementRepository.HasDateOverlap(newProposal);
+
+        if (hasConflict)
+        {
+            return Conflict("There is already a Transfer Agreement with this company tin within the selected date range");
+        }
+
+        await repository.AddTransferAgreementProposal(newProposal);
+
+        return CreatedAtAction(nameof(GetTransferAgreementProposal), new { id = newProposal.Id }, newProposal);
+    }
+
+    /// <summary>
+    /// Get TransferAgreementProposal by Id
+    /// </summary>
+    /// <param name="id">Id of TransferAgreementProposal</param>
+    /// <response code="200">Successful operation</response>
+    /// <response code="400">You cannot Accept/Deny your own TransferAgreementProposal, you cannot Accept/Deny a TransferAgreementProposal for another company or this proposal has run out</response>
+    /// <response code="404">TransferAgreementProposal expired or deleted</response>
+    [ProducesResponseType(typeof(TransferAgreementProposal), 200)]
+    [ProducesResponseType(typeof(void), 400)]
+    [ProducesResponseType(typeof(void), 404)]
+    [HttpGet("{id}")]
+    public async Task<ActionResult<TransferAgreementProposal>> GetTransferAgreementProposal(Guid id)
+    {
+        var proposal = await repository.GetNonExpiredTransferAgreementProposal(id);
+
+        if (proposal == null)
+        {
+            return NotFound("TransferAgreementProposal expired or deleted");
```

Comment:
According to our guidelines 404 must not contain a body

Reply:
Corrected